doc/schema/attribute-types/integer.md

```diff
@@ -1,62 +1,85 @@
-# integer({min, max, references, default, required, autoIncrement})
+# integer({autoIncrement, min, max, step, primary, required, default, unique})
 
 Defines an attribute as being an integer.
 
 ```ts
-export const Todo: PartialSchema = {
+export const Todo = {
   name: "Todo",
   attributes: {
     importance: integer({ required: true }),
-  }
-}
+  },
+} satisfies PartialSchema
 ```
 
 ## Parameters
 
-- `autoIncrement` [{Boolean=false}] - If `true`, increments the field. Example: `integer({autoIncrement: true})`
-- `default` [{Integer}] - The default value of the attribute.  Example: `integer({default: 10})`
-- `max` [{Integer=Infinity}] - The maximum number allowed. Defaults to Infinity. Example: `integer({max: 1000})`
-- `min` [{Integer=-Infinity}] - The minimum number of characters allowed.  Defaults to -Infinity. Example: `integer({min: 1})`
-- `references` - [See References]()
-- `required` [{Boolean=false}] - If the attribute must be provided.
-- `step` [{Integer=1}] - If the granularity the value must adhere to. This number must be an integer itself. Example: `integer({step: 2})`
+| key             | description                                                                         |   type    | optional |   default   |
+| --------------- | ----------------------------------------------------------------------------------- | :-------: | :------: | :---------: |
+| `unique`        | If the value must be unqiue. <br/> Example: `integer({unique: true})`               | `Boolean` |   Yes    |   `false`   |
+| `default`       | The default value of the attribute. <br/> Example: `integer({default: 0})`          | `Integer` |   Yes    |   `null`    |
+| `required`      | If the attribute must be provided. <br/> Example: `integer({required: true})`       | `Boolean` |   Yes    |   `false`   |
+| `primary`       | If the attribute is a primary key. <br/> Example: `integer({primary: true})`        | `Boolean` |   Yes    |   `false`   |
```

Comment:
the first 4 (unique, default, required, primary) are a base data type that is extended by all types - should we duplicate them in each attribute doc file or just the unique ones, in this case: step, max, min, autoIncrement

Reply:
I would duplicate them for now. At some point in the future ideally, these would be JSDoc strings on properties of a type union (or similar) and would be documented in all these locations based on a single shared type.